skrub/_joiner.py

```diff
@@ -3,29 +3,25 @@
 """
 
 import numpy as np
-import pandas as pd
 from sklearn.base import BaseEstimator, TransformerMixin, clone
 from sklearn.compose import make_column_transformer
 from sklearn.feature_extraction.text import HashingVectorizer, TfidfTransformer
 from sklearn.pipeline import make_pipeline
-from sklearn.preprocessing import FunctionTransformer, StandardScaler
+from sklearn.preprocessing import StandardScaler
 from sklearn.utils.validation import check_is_fitted
 
-from skrub import _join_utils, _matching, _utils
-from skrub._dataframe._namespace import is_pandas, is_polars
-from skrub._datetime_encoder import DatetimeEncoder
-
+from . import _dataframe as sbd
+from . import _join_utils, _matching, _utils
 from . import _selectors as s
+from ._check_input import CheckInputDataFrame
+from ._datetime_encoder import DatetimeEncoder
+from ._to_str import ToStr
 from ._wrap_transformer import wrap_transformer
 
-
-def _as_str(column):
-    return column.fillna("").astype(str)
-
-
 DEFAULT_STRING_ENCODER = make_pipeline(
-    FunctionTransformer(_as_str),
+    ToStr(),
```

Comment:
```suggestion
    FunctionTransformer(partial(sbd.fill_nulls, value="")),
    ToStr(),
```

you will need `from functools import partial` and to restore the `FunctionTransformer` import